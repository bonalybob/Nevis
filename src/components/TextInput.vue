<template>
  <div
    class="flex w-full px-2 py-2 border-solid border-b border-blue-point4 last:border-b-0"
  >
    <label
      :for="label"
      class="font-body font-semibold text-blue flex-none px-2 select-none"
      >{{ label }}</label
    >
    <input
      :name="label"
      :value="value"
      :type="inputType"
      class="font-body outline-none flex-1 appearance-none"
      @input="$emit('input', $event.target.value)"
    />
  </div>
</template>

<script>
export default {
  name: 'DropdownInput',

  props: {
    value: {
      default: '',
    },
    label: {
      type: String,
      default: '',
    },
    type: {
      type: String,
      default: '',
    },
  },

  computed: {
    inputType: function () {
      if (this.type === 'password') return 'password'
      else return 'text'
    },
  },
}
</script>
