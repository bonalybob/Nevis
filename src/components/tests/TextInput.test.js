import { mount } from '@vue/test-utils'
import TextInput from '@/components/TextInput'

test('Is a Vue Instance', () => {
  const wrapper = mount(TextInput)
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(TextInput, {
    propsData: {},
  })
  expect(wrapper.element).toMatchSnapshot()
})

test('Assigns Correct Input Type', () => {
  const wrapper = mount(TextInput, {
    propsData: {
      type: 'password',
    },
  })
  expect(wrapper.vm.inputType).toBe('password')
  wrapper.setProps({ type: '' })
  expect(wrapper.vm.inputType).toBe('text')
  wrapper.setProps({ type: 'text' })
  expect(wrapper.vm.inputType).toBe('text')
})
