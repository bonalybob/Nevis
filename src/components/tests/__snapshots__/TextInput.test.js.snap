// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div
  class="flex w-full px-2 py-2 border-solid border-b border-blue-point4 last:border-b-0"
>
  <label
    class="font-body text-blue flex-none px-2 select-none"
    for=""
  >
    
  </label>
   
  <input
    class="font-body outline-none flex-1 appearance-none"
    name=""
    type="text"
  />
</div>
`;
